test/cusparse/interfaces.jl

```diff
@@ -319,37 +319,21 @@ using LinearAlgebra, SparseArrays
         @test SparseMatrixCSC(CuSparseMatrixCSR(T)) ≈ f(S)
     end
 
-    VERSION >= v"1.7" && @testset "UniformScaling with $typ($dims)" for
-            typ in [CuSparseMatrixCSR, CuSparseMatrixCSC],
-            dims in [(10, 10), (5, 10), (10, 5)]
-        S = sprand(Float32, dims..., 0.1)
-        dA = typ(S)
-
-        @test Array(dA + I) == S + I
-        @test Array(I + dA) == I + S
-
-        @test Array(dA - I) == S - I
-        @test Array(I - dA) == I - S
-    end
-
-    @testset "Diagonal with $typ(10, 10)" for
-        typ in [CuSparseMatrixCSR, CuSparseMatrixCSC]
-
-        S = sprand(Float32, 10, 10, 0.8)
-        D = Diagonal(rand(Float32, 10))
-        dA = typ(S)
-        dD = adapt(CuArray, D)
-
-        @test Array(dA + dD) == S + D
-        @test Array(dD + dA) == D + S
-
-        @test Array(dA - dD) == S - D
-        @test Array(dD - dA) == D - S
-
-        @test dA + dD isa typ
-        @test dD + dA isa typ
-        @test dA - dD isa typ
-        @test dD - dA isa typ
+    VERSION >= v"1.7" && @testset "UniformScaling and Diagonal" begin
```

Comment:
This isn't equivalent; some of these tests were also being run on <1.7, and testing with `Diagonal(rand)` is more general than just using `I`.

Reply:
Ok, I'm including the case of Diagonal(rand), and excluding the tests of UniformScaling for v < 1.7. But, what happens if I remove VERSION >= v"1.7"? UniformScaling and Diagonal should be working also on previous version of Julia. Am I wrong?

Reply:
IIRC I added that version check because the implementation relies on broadcast, and sparse broadcast is only available on Julia 1.7+.